cassandra/metadata.py

```diff
@@ -1598,9 +1598,13 @@ def _query_build_row(self, query_string, build_func):
 
     def _query_build_rows(self, query_string, build_func):
         query = QueryMessage(query=query_string, consistency_level=ConsistencyLevel.ONE)
-        response = self.connection.wait_for_response(query, self.timeout)
-        result = dict_factory(*response.results)
-        return [build_func(row) for row in result]
+        responses = self.connection.wait_for_responses((query), timeout=self.timeout, fail_on_error=False)
+        (success, response) = responses[0]
+        if success:
+            result = dict_factory(*response.results)
+            return [build_func(row) for row in result]
+        else:
+            return []
```

Comment:
One minor question: curious if you thought about handling the exception by type [like we do on the main query path](https://github.com/datastax/python-driver/blob/3.6.0/cassandra/metadata.py#L2023-L2027) (this can prevent it swallowing different kinds of unexpected errors).